reporters/kamon-newrelic/src/main/scala/kamon/newrelic/NewRelicConfig.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2013-2020 The Kamon Project <https://kamon.io>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kamon.newrelic
+
+import com.typesafe.config.Config
+import kamon.newrelic.NewRelicConfig.NewRelicApiKey
+import kamon.newrelic.NewRelicConfig.NewRelicApiKey.{InsightsInsertKey, LicenceKey}
+
+import java.net.URL
+import java.time.Duration
+
+
+private case class NewRelicConfig(
+                                   apiKey: NewRelicApiKey,
+                                   enableAuditLogging: Boolean,
+                                   userAgent: String,
+                                   callTimeout: Duration,
+                                   spanIngestUri: Option[URL],
+                                   metricIngestUri: Option[URL]
+                                 )
+
+private object NewRelicConfig {
+
+  def fromConfig(config: Config): NewRelicConfig = {
+    val nrConfig = config.getConfig("kamon.newrelic")
+
+    // TODO maybe some validation around these values?
+    val apiKey = if (nrConfig.hasPath("licence-key")) {
```

Comment:
u mean `licence-key = ""`?

Reply:
if yes, then it is imo fine, someone just want to use the licence path. remember that the `metric-ingest-uri = "https://metric-api.newrelic.com/metric/v1"` can be overridden. so if someone uses mock then it is ok to have `""`

Reply:
Plus one to that, I'd only use it if it's non empty. I'm also wondering if we shouldn't go the opposite direction and try using `licence-key` only when `nr-insights-insert-key` is not present (or present but empty?). This would improve the backwards compatibility for clients.

Reply:
Yeah, I like @majk-p idea

Reply:
if someone does `license-key = ${?NOT_EXISTING_ENV}` then `hasPath` returns `false`

Reply:
I would not expect that every lib user has such approach.

Those keys are present in [reference.conf](https://github.com/kamon-io/Kamon/blob/master/reporters/kamon-newrelic/src/main/resources/reference.conf). I'd expect them to be both there.

Reply:
`nr-insights-insert-key` is never absent because it is defined here https://github.com/kamon-io/Kamon/blob/master/reporters/kamon-newrelic/src/main/resources/reference.conf#L8. so i can't make logic _"try using licence-key only when nr-insights-insert-key is not present"_ that is backwards compatible